### PR TITLE
feat(backups): implement SCB restore

### LINKED ISSUE #2082: I should be able to recover channel balances by importing a SCB file as part of the wallet recovery process
## Detailed Description

As a counterpart to https://github.com/LN-Zap/zap-desktop/issues/1971 I should be able to restore channel balances as part of a wallet recovery.

### LINKED ISSUE #2217: Use tree directory structure to store backups
## Detailed Description
Right now backups are stored in Google Drive root. Tree directory structure needs to be worked through and implemented to simplify management and reduce potential collisions with other platforms. The same structure should be applied to Dropbox and local backup strategies.

### PR DESCRIPTION
**Pull request by korhaliv**
## Description:
Resolves #2082 
Resolves #2217 

## How Has This Been Tested?
Manually

## Types of changes:
New feature

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [ ] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: electron/walletBackup/local/index.js [hunk 1]
```diff
@@ -27,8 +42,9 @@ export default class BackupService {
    * @memberof BackupService
    */
   saveBackup = chainify(async ({ walletId, fileId: dir, backup }) => {
-    const filePath = `${dir}/${walletId}`
-    await writeFileAsync(filePath, backup)
+    const filePath = path.join(dir, walletId)
+    await mkdirAsync(filePath, { recursive: true })
+    await writeFileAsync(path.join(filePath, config.backup.filename), backup)
     return dir
```
Comment (mrfelton) [inline electron/walletBackup/local/index.js]: Should this really be returning `dir`? Ii looks like the caller is expecting a file id. Comments also suggest this.

File: electron/walletBackup/service.js [hunk 2]
```diff
@@ -44,6 +44,33 @@ export default function createBackupService(mainWindow) {
...
+          fileId: backupId,
+        })
+        send('queryWalletBackupSuccess', { backup, walletId })
+      }
+    } catch (e) {
+      send('queryWalletBackupFailure', { walletId })
+      mainLog.warn(
+        `Unable to query backup for wallet %s using provider %s:  %o`,
+        walletId,
+        provider
```
Comment (mrfelton) [inline electron/walletBackup/service.js]: I think you forgot to pass in `e` for use with the `%o` placeholder

File: renderer/reducers/backup.js [hunk 3]
```diff
@@ -224,6 +274,54 @@ export const setBackupPathLocal = localPath => {
     localPath,
   }
 }
+/**
+ * setRestoreMode - sets if we are in the restore mode currently.
+ *
+ * @param {boolean} value local filesystem directory URI
```
Comment (mrfelton) [inline renderer/reducers/backup.js]: `local filesystem directory URI` seems like the wrong description for this

File: renderer/reducers/backup.js [hunk 4]
```diff
@@ -224,6 +274,54 @@ export const setBackupPathLocal = localPath => {
...
+  try {
+    const grpc = await grpcService
+    const result = await grpc.services.Lightning.restoreChannelBackups({
+      multi_chan_backup: backup,
+    })
+    // TODO add intl support
+    dispatch(showNotification(`Wallet backup imported successfully`))
+    return result
+  } catch (e) {
+    dispatch(showError(`Backup import has failed: incorrect backup format`))
```
Comment (mrfelton) [inline renderer/reducers/backup.js]: Can this fail for other reasons? Should we include `e.message` in this to help people debug import failures?

File: electron/walletBackup/service.js (hunk 2 above)
Comment (korhaliv) [reply electron/walletBackup/service.js]: ah, good catch. i was wondering why i see undefined errors on the console

Comment (korhaliv) [reply electron/walletBackup/service.js]: done

File: electron/walletBackup/local/index.js (hunk 1 above)
Comment (korhaliv) [reply electron/walletBackup/local/index.js]: renamed backupId->locationHint

File: renderer/reducers/backup.js (hunk 3 above)
Comment (korhaliv) [reply renderer/reducers/backup.js]: done

File: renderer/reducers/backup.js (hunk 4 above)
Comment (korhaliv) [reply renderer/reducers/backup.js]: added message. We had this convo is slack. eventually, we'd want customized messages here

File: config/default.js [hunk 5]
```diff
@@ -123,6 +123,7 @@ module.exports = {
     // if false, autopilot selection won't be available
     mainnetAutopilot: false,
     networkSelection: false,
+    scbRestore: true,
```
Comment (mrfelton) [inline config/default.js]: I think the default should be `false` here

Comment (korhaliv) [reply config/default.js]: updated

**Comment by korhaliv** [comment]
@mrfelton test comment

**Pull request merged**